### PR TITLE
Nwd nnw

### PR DESCRIPTION
**Pull request by opiotrowsky**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: nwd-nnw
🏅 Granted 6 XP for @opiotrowsky

**Comment by ziobron** [comment]
⏱ 3 XP extra for @opiotrowsky for on-time delivery

File: homework/nwd-nnw/nwdNww.hpp [hunk 1]
```diff
@@ -1,11 +1,45 @@
...
+        num0 = lhs;
+        lhs = rhs;
+        rhs = num0;
+    }
+    if(rhs == 0) {
+        return lhs;
+    }
+    if(lhs % rhs == 0) {
+        return rhs;
+    }
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:22]: Use std::swap instead create new variable

if (lhs > rhs) {
   std::swap(lhs, rhs)
}

you can also use abs

File: homework/nwd-nnw/nwdNww.hpp [hunk 2]
```diff
@@ -1,11 +1,45 @@
...
+        lhs *= (-1);
+    }
+    if(rhs < 0) {
+        rhs *= (-1);
+    }
+    if(lhs < rhs) {
+        num0 = lhs;
+        lhs = rhs;
+        rhs = num0;
+    }
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:16]: you duplicate code here. Check loop below

File: homework/nwd-nnw/nwdNww.hpp [hunk 3]
```diff
@@ -1,11 +1,45 @@
...
-    // TODO: Implement me :)
-    return -1;
+
+    int num0, num1;
+    if(lhs < 0){
+        lhs *= (-1);
+    }
+    if(rhs < 0) {
+        rhs *= (-1);
+    }
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:11]: use std::abs

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**